### PR TITLE
add utils and tests

### PR DESCRIPTION
**Pull request by digitalsadhu**
This PR adds utils in prep for server changes to how publishing is done.

* `GET /:org/pkg/:name` is used to get back a list of versions which is expected to look like:
```json
{ "versions": [
    [1, {"version": "1.3.2"}],
    [2, {"version": "2.1.1"}]
] }
```
* `GET /:org/pkg/:name/:version` is then used to fetch integrity hashes and expects:
```js
{
  version: '8.4.1',
  name: 'fuzz',
  org: 'biz',
  files: [
    {
      integrity: 'sha512-T2qS6EBvOIu10bhUas3FhD39KkwIiXxplJ13q2EdXcA7nlYljlLKaymKhqz49f7qrEKhdISc4q5N+bk0Y1Y/NA==',
      pathname: '/main/index.js',
      mimeType: 'application/javascript',
      type: 'file',
      size: 75208
    },
    {
      integrity: 'sha512-0K6U6pmI04xIBGE+KgfSRNMY0gBmKAwjWzZ+DM/tkicZSG+Uz5erTFw1Zru/0wXUPs256glMX24n0f1Q4z62tw==',
      pathname: '/main/index.js.map',
      mimeType: 'application/json',
      type: 'file',
      size: 771205
    },
  ]
}
```

### CODE REVIEW

File: utils/calculate-file-hash.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const fs = require('fs');
+
+module.exports = path =>
+    new Promise((resolve, reject) => {
+        const hash = crypto.createHash('sha512');
+        const rs = fs.createReadStream(path);
+        rs.on('error', reject);
+        rs.on('data', chunk => hash.update(chunk));
+        rs.on('end', () => resolve(hash.digest('base64')));
+    });
```
Comment (trygve-lie) [inline utils/calculate-file-hash.js]: We can do this as follow:

```
const hash = crypto.createHash('sha512');
const file = fs.createReadStream(path);

pipeline(file, hash, error => {
    if (error) return reject(error);
    resolve(hash.digest('base64')));
});

File: utils/calculate-files-hash.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const crypto = require('crypto');
+const fileHash = require('./calculate-file-hash');
```
Comment (trygve-lie) [inline utils/calculate-files-hash.js]: Any reason why this file is requiring it self?

Comment (digitalsadhu) [reply utils/calculate-files-hash.js]: umm... whoops?

Comment (digitalsadhu) [reply utils/calculate-files-hash.js]: oh, no its not. `calculate-files-hash` is requiring `calculate-file-hash`. The first is plural

Comment (trygve-lie) [reply utils/calculate-files-hash.js]: Ahh..

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macOS-latest, windows-latest]
-        node-version: [10.x, 12.x]
+        node-version: [12.x]
```
Comment (trygve-lie) [inline .github/workflows/nodejs.yml]: Test on 13 also?

**Pull request merged**